// @netlify-tools/core/lib/utils/netlify.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Netlify = void 0;
const NetlifyAPI = require('netlify');
class Netlify {
    constructor(configuration) {
        this.configuration = configuration;
        this.client = new NetlifyAPI(configuration.accessToken);
    }
    async getContextSettings() {
        var _a, _b;
        const { build_settings: buildSettings } = await this.client.getSite({
            site_id: this.configuration.siteId,
        });
        return {
            variables: (_a = buildSettings.env) !== null && _a !== void 0 ? _a : [],
            branches: (_b = buildSettings.allowed_branches) !== null && _b !== void 0 ? _b : [],
        };
    }
    async updateContextSettings(payload) {
        const buildSettings = {};
        if (payload.variables) {
            buildSettings.env = payload.variables;
        }
        if (payload.branches) {
            buildSettings.allowed_branches = payload.branches;
        }
        const { build_settings: updatedBuildSettings, } = await this.client.updateSite({
            site_id: this.configuration.siteId,
            body: {
                build_settings: buildSettings,
            },
        });
        return {
            variables: updatedBuildSettings.env,
            branches: updatedBuildSettings.allowed_branches,
        };
    }
}
exports.Netlify = Netlify;
